### PR TITLE
Enable hermetic builds wip dnm

### PR DESCRIPTION
**Pull request by simonbaird**
Not sure if this will work, but let's try it.

Ref: https://issues.redhat.com/browse/EC-360

### CODE REVIEW

**Comment by simonbaird** [comment]
Build pipeline still shows prefetch being skipped, so this patch is not sufficient..

**Comment by simonbaird** [comment]
I was tricked by the key ordering. Also I didn't understand what `prefetch-input` was for. (Now it's set to `gomod`.)

Pushed a new revision.

File: .tekton/cli-main-ci-pull-request.yaml [hunk 1]
```diff
@@ -100,11 +100,11 @@ spec:
       description: Skip checks against built image
       name: skip-checks
       type: string
-    - default: "false"
+    - default: "true"
```
Comment (zregvart) [inline .tekton/cli-main-ci-pull-request.yaml:103]: Why change the defaults? Set the parameter in the `spec.params` above

**Review by zregvart — COMMENTED** [review]
Would it make sense to use the cosign binary from the [RHTAS cosign image](https://catalog.redhat.com/software/containers/rhtas/cosign-rhel9/65f1eed5b8a552c8d7ee4c53?tag=1.0.0-1712564506&architecture=amd64&image=6613aad2c8d30b87f4c93f82)?

**Comment by simonbaird** [comment]
Pushed a POC DNM commit to use the downstream cosign image to get a copy of the cosign binary.

File: .tekton/cli-main-ci-pull-request.yaml (hunk 1 above)
Comment (simonbaird) [reply .tekton/cli-main-ci-pull-request.yaml]: Oh yeah, thanks.

**Comment by simonbaird** [comment]
Currently still fails on the `microdnf install` command.

File: Dockerfile.dist [hunk 2]
```diff
@@ -82,8 +99,10 @@ LABEL \
   com.redhat.component="ec-cli"
 
 # Install cosign and other tools we want to use in the Tekton task
+# Todo: How to prefetch these rpms in the prefetch-dependencies task??
```
Comment (zregvart) [inline Dockerfile.dist:102]: Watch [this space](https://github.com/containerbuildsystem/cachi2/discussions/465)

**Comment by simonbaird** [comment]
Replaced by #1762.

**Pull request closed without merging**